packages/data-api/src/EventsFetchQuery.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { DataFetchQuery } from './DataFetchQuery';
+
+export class EventsFetchQuery extends DataFetchQuery {
+  constructor(database, options) {
+    super(database, options);
+
+    // immutable options specific to event fetching
+    const { eventId, dataGroupCode, dataElementCodes } = options;
+    this.eventId = eventId;
+    this.dataGroupCode = dataGroupCode;
+    this.hasDataElements = dataElementCodes && dataElementCodes.length > 0;
+  }
+
+  getAliasedColumns() {
+    const aliasedColumns = [
+      'date',
+      'entity_code AS "entityCode"',
+      'entity_name AS "entityName"',
+      'event_id as "eventId"',
+      'value',
+      'type',
+    ];
+    if (this.hasDataElements) {
+      aliasedColumns.push('data_element_code AS "dataElementCode"');
+    }
+    return aliasedColumns.join(', ');
+  }
+
+  getInnerJoins() {
+    const joins = [this.createInnerJoin(this.entityCodes, 'entity_code')];
+    if (this.hasDataElements) {
+      joins.push(this.createInnerJoin(this.dataElementCodes, 'data_element_code'));
+    }
+    return joins.join('\n');
+  }
+
+  getWhereClause() {
+    const conditions = [];
+    const params = [];
+
+    if (this.startDate) {
+      conditions.push('date >= ?');
+      params.push(this.startDate);
+    }
+    if (this.endDate) {
+      conditions.push('date <= ?');
+      params.push(this.endDate);
+    }
+    if (this.dataGroupCode) {
+      conditions.push('data_group_code = ?');
+      params.push(this.dataGroupCode);
+    }
+    if (this.eventId) {
+      conditions.push('event_id = ?');
+      params.push(this.eventId);
+    }
+
+    if (conditions.length === 0) return '';
```

Comment:
```suggestion
    if (conditions.length === 0) {
      return '';
    }
```
Totally optional and correct me if I'm wrong, but I thought we tended to avoid omiting brackets after if clauses?

Reply:
Hmmm, well I think opinion is actually divided there!

I personally prefer using the brackets, but at some point there was pushback and we disabled the eslint rule, so now I have a bias towards brackets but am not consistent and don't really care.

Tupaia's second developer, Marty, was hugely pro the brackets being there, which is what got me in the habit and probably why the eslint rule used to exist.

Either way I'm not fussed!